oeps/oep-0005.rst

```diff
@@ -0,0 +1,263 @@
+   or have to maintain compatibility with multiple database versions until all
+   services are ready to update. This hinders independent development on the
+   services.
+
+5. Because all of the services are installed on the same image, the image
+   either has to be updated by every user after downloading (to update to the
+   latest code versions), or the image has to be rebuilt every time any of the
+   installed services change (which results in lots of churn).
+
+6. Because the `Vagrant`_ image contains many separate services, the definition
+   of that image has to be separate from any of the service repositories, which
+   makes it hard to update (and to push updates out to users).
+
+
+Specification
+=============
+
+Interface
+~~~~~~~~~
+
+The primary developer interaction with the local development environment will
+be through ``make`` shortcuts in the repository containing the code for the
+service being developed.  For example, in order to work on an ``edx-platform``
+local development environment, a developer could run ``make dev.watch`` from a
+checked-out `edx-platform`_ repository.
+
+The following targets will be standard::
+
+    make dev.watch  # Start the local development environment services (if
+                    # they aren't already) and display their logs on stdout
+
+    make dev.shell  # Start the local development environment services (if they
+                    # aren't already) and access a shell on the primary
+                    # container (as the user that owns the service, in the
+                    # directory that has the service source code)
+
+    make dev.clean  # Stop the local development environment services and
+                    # remove any intermediate data that they stored in the
+                    # .dev/ directory
+
+    make dev.*      # Start the local development environment services (if they
+                    # aren't already) and run `make *` from the code directory
+                    # of the primary service (as the service owner)
+
+Implementation
+~~~~~~~~~~~~~~
+
+Requirements
+------------
+
+1. Third-party services like MySQL and MongoDB should be run using vendor
+   provided official containers. If no reasonable alternative exists to
+   customizing a provided container, the minimal changes should be applied as a
+   layer on the provided container.
+
+   **Rationale**: We should be aiming to minimize the amount of work needed to
+   support our development environments. Minimizing changes to third-party
+   containers means we have less code/configuration that we need to update as
+   third-party containers and applications change. Minimizing customization
+   should also keep push us in the direction of making it easy to deploy our
+   stack, because there will be fewer customizations needed by operators of
+   Open edX in production.
+
+2. Open edX containers must run all of their processes via `supervisor`_ to
```

Comment:
No. The goal of OEP-5 is to make the development environment run faster while still mirroring our production deployment as much as is reasonable. In production, nginx is deployed to each of the webhosts. As such, we should be deploying it to the same container in development mode, so that it's easy to validate that all of the managed configuration is correct.

If, in the future, we get to a point where nginx is deployed on a separate host, then I would recommend we move it to a second container.